worker/include/Utils.hpp

```diff
@@ -356,26 +356,36 @@ namespace Utils
 	template<typename T>
 	class ObjectPoolAllocator
 	{
+		thread_local static std::shared_ptr<std::vector<T*>> pool_data;
```

Comment:
Why does it need a smart pointer?

Reply:
@nazar-pc  I found logical problem with vector of pointers.
When ```ObjectPoolAllocator<T>(const ObjectPoolAllocator<U>& other)``` called I copied vector, but should copy reference to it. Otherwise, the destructor of ```ObjectPoolAllocator``` will clean up memory pointers multiple times.